tests/clone-flags.c

```diff
@@ -23,11 +23,26 @@
 static const int child_exit_status = 42;
 static pid_t pid;
 
+static void
+retrieve_userns(pid_t pid, char *userns_buf, size_t userns_buflen)
+{
+	char *fname = xasprintf("/proc/%d/ns/user", pid);
+	int rc = readlink(fname, userns_buf, userns_buflen - 1);
+	if ((unsigned int) rc >= userns_buflen)
+		perror_msg_and_fail("readlink");
+	userns_buf[rc] = '\0';
```

Comment:
It's not the reason to fail the whole test if this readlink failed.  Let's do the following instead:
```
        if (rc < 0) {
                perror(fname);
                userns_buf[0] = '\0';
        } else if ((size_t) rc >= userns_buflen)
                error_msg_and_fail("readlink: %s: result is too long", fname);
        else
                userns_buf[rc] = '\0';
```

Reply:
fixed locally.